### DISCUSSION TITLE
condition checking for on enter

### DISCUSSION THREAD

**Discussion by daggs1** [endless-sky/endless-sky#9661] [General]
Greetings,

I'm working on a mission that requires passing via 5 way points.
I want to have an event happening at the enter to system 5 iff system 1-4 were accessed before.
I understand I can use the condition++ for systems 1-4 but afaik, there is no way to test the condition on enter.
any ideas if it can be done?
testing the condition on systems 1-4 and setting the event is an acceptable solution too

Thanks

**Comment by Amazinite** [comment]
If the systems are waypoints, I believe the 'on waypoint' action will run when all waypoints have been visited. Otherwise, you'd need to have a conversation appear for each 'on enter'. The conversation could contain action and branch nodes for setting and testing the condtions for the number of systems that have been visited and triggering an event once all systems have been visited.

**Reply by daggs1** [reply]
so conditions are evaluated on every system switch in this case as there are no landable planets in the system in question?
if so, can that same logic be applied for unhiding systems?
